scss/mixins/_forms.scss

```diff
@@ -79,6 +79,7 @@
     &.is-#{$state} {
       ~ .custom-control-indicator {
         background-color: rgba($color, .25);
+        box-shadow: 0 0 0 1px #fff, 0 0 0 3px #dc3545;
```

Comment:
Because this ruleset is in a Sass mixin, and we're passing parameters for the color, you'll want to tweak this line just a bit. Rather than hardcode the hex colors, use this:

```scss
box-shadow: 0 0 0 1px $body-bg, 0 0 0 3px $color;
```

This way, if you customize the `$body-bg` variable, it'll get picked up automatically here. And when we use this mixin to generate the valid and invalid styles, that color will be correct, too.

Reply:
After you make that change, we can move this to the focus state.

```scss
.was-validated &:#{$state},
&.is-#{$state} {
  ~ .custom-control-indicator {
    background-color: rgba($color, .25);
  }
  &:focus {
    ~ .custom-control-indicator {
      box-shadow: 0 0 0 1px $body-bg, 0 0 0 3px $color;
    }
  }
}
```

Reply:
Thank you @mdo I will make these changes send another commit. This is so helpful!